pymc3/distributions/timeseries.py

```diff
@@ -464,6 +468,74 @@ def logp(self, x):
     def _distr_parameters_for_repr(self):
         return ["mu", "cov"]
 
+    def random(self, point=None, size=None):
+        """
+        Draw random values from MvGaussianRandomWalk.
+
+        Parameters
+        ----------
+        point: dict, optional
+            Dict of variable values on which random values are to be
+            conditioned (uses default point if not specified).
+        size: int or tuple of ints, optional
+            Desired size of random sample (returns one sample if not
+            specified).
+
+        Returns
+        -------
+        array
+
+
+        Examples
+        -------
+        Create one sample from a 2-dimensional Gaussian random walk with 10 timesteps::
+
+            mu = np.array([1.0, 0.0])
+            cov = np.array([[1.0, 0.0], [0.0, 2.0]])
+            sample = MvGaussianRandomWalk(mu, cov, shape=(10, 2)).random()
+
+        Create three samples from a 2-dimensional Gaussian random walk with 10 timesteps::
+
+            mu = np.array([1.0, 0.0])
+            cov = np.array([[1.0, 0.0], [0.0, 2.0]])
+            sample = MvGaussianRandomWalk(mu, cov, shape=(10, 2)).random(size=3)
+
+        Create four samples from a 2-dimensional Gaussian random walk with 10
+        timesteps, indexed with a (2, 2) array::
+
+            mu = np.array([1.0, 0.0])
+            cov = np.array([[1.0, 0.0], [0.0, 2.0]])
+            sample = MvGaussianRandomWalk(mu, cov, shape=(10, 2)).random(size=(2, 2))
+        """
+
+        # for each draw specified by the size input, we need to draw time_steps many
+        # samples from MvNormal.
+
+        size = to_tuple(size)
+        multivariate_samples = self.innov.random(point=point, size=size)
+        # this has shape (size, self.shape)
+
+        if len(self.shape) >= 2:
```

Comment:
We cannot have more than 2 dimensions for self.shape because Theano does not support cholesky decomposition with batches of covariances matrices. 

```suggestion
        if len(self.shape) == 2
```

Reply:
Thanks, @Sayam753! I really like how nice the code is turning out. I'll try to work on getting the tests for RV `mu` and others. I suspect it will need a slight modification of `generate_shapes`.